OpenRA.Mods.Common/Activities/ExternalCaptureActor.cs

```diff
@@ -44,7 +51,16 @@ public override Activity Tick(Actor self)
 				return NextActivity;
 			}
 
-			var nearest = target.Actor.OccupiesSpace.NearestCellTo(mobile.ToCell);
```

Comment:
public Target UpdateForAttacker(Actor attacker)
With my `Target.UpdateForAttacker` suggestion above this can then simplify to:
```csharp
var newTarget = target.UpdateForAttacker(self);
if (newTarget != target)
	return new ExternalCaptureActor(self, newTarget);
```

This is a much shorter and cleaner pattern, which is more suitable to use across all the activities that will need it.